nexus/db-queries/src/db/datastore/fm.rs

```diff
@@ -139,10 +151,181 @@ impl DataStore {
+                }
+            }
+
+            map
+        };
+        // Next, load the case metadata entries and marry them to the sets of
+        // ereports assigned to those cases that we loaded in the previous step.
+        let cases = {
+            let mut cases = iddqd::IdOrdMap::new();
+            let mut paginator =
+                Paginator::new(SQL_BATCH_SIZE, PaginationOrder::Descending);
+            while let Some(p) = paginator.next() {
+                let batch = self
+                    .fm_sitrep_cases_list_on_conn(
+                        id,
+                        &p.current_pagparams(),
+                        &conn,
+                    )
+                    .await
+                    .map_err(|e| {
+                        e.internal_context("failed to list sitrep cases")
+                    })?;
+                paginator = p.found_batch(&batch, &|case| case.id);
+                cases.extend(batch.into_iter().map(|case| {
+                    let model::fm::CaseMetadata {
+                        id,
+                        sitrep_id: _,
+                        created_sitrep_id,
+                        closed_sitrep_id,
+                        comment,
+                        de,
+                    } = case;
+                    let id = id.into();
+
+                    // Take all the case ereport assignments we've collected for this case.
+                    let ereports = case_ereports
+                        .remove(&id)
+                        // If there's no entry in the map of case ereport
+                        // assignments for this case, then that just means that the
+                        // case has no ereports assigned to it, so insert an empty
+                        // map here.
+                        .unwrap_or_default();
+                    fm::Case {
+                        id,
+                        created_sitrep_id: created_sitrep_id.into(),
+                        closed_sitrep_id: closed_sitrep_id.map(Into::into),
+                        de: de.into(),
+                        comment,
+                        ereports,
+                    }
+                }));
+            }
 
-        Ok(Sitrep { metadata })
+            cases
+        };
+
+        Ok(Sitrep { metadata, cases })
+    }
+
+    async fn fm_sitrep_cases_list_on_conn(
+        &self,
+        sitrep_id: SitrepUuid,
+        pagparams: &DataPageParams<'_, DbTypedUuid<CaseKind>>,
+        conn: &async_bb8_diesel::Connection<DbConnection>,
+    ) -> ListResultVec<model::fm::CaseMetadata> {
+        paginated(case_dsl::fm_case, case_dsl::id, &pagparams)
+            .filter(case_dsl::sitrep_id.eq(sitrep_id.into_untyped_uuid()))
+            .select(model::fm::CaseMetadata::as_select())
+            .load_async::<model::fm::CaseMetadata>(conn)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
+    }
+
+    fn fm_sitrep_read_ereports_query(
```

Comment:
I know it took a ton of thought and effort (and db restructuring) to get here, but this ended up looking very simple - nicely done.